test/evaluate/test_folding.sh

```diff
@@ -52,12 +48,7 @@ actual_warnings=$temp/actwarnings.log
 expected_warnings=$temp/expwarnings.log
 warning_diffs=$temp/warnings.diff
 
-if ! ( cd $temp; $CMD $src ) > $src1 2> $messages # compile, dumping symbols
-then
-  cat $messages
-  echo FAIL compilation
-  exit 1
-fi
+( cd $temp; $CMD $src ) > $src1 2> $messages # compile, dumping symbols
```

Comment:
The test is supposed to fail if there are compilation errors. Does that still happen?

Reply:
This code can also probably return, as per my previous comment, it complained messages didn't exist, so couldn't `cat` it, yet it needed to exist in order to pass the clause.